src/test/java/io/supertokens/test/session/api/SessionGetAPIJWTTest.java

```diff
@@ -0,0 +1,164 @@
+                .sendGETRequest(process.getProcess(), "", "http://localhost:3567/recipe/session", map, 1000, 1000,
+                        null,
+                        Utils.getCdiVersionLatestForTests(), Session.RECIPE_ID);
+        assertEquals(response.entrySet().size(), 2);
+        assertEquals(response.get("status").getAsString(), "UNAUTHORISED");
+
+        assertEquals(response.get("message").getAsString(), "Session does not exist.");
+    }
+
+    @Test
+    public void testThatGetWithValidParametersWorks() throws Exception {
+        String[] args = {"../"};
+
+        String sessionJsonInput = "{\n" +
+                "\t\"userId\": \"UserID\",\n" +
+                "\t\"userDataInJWT\": {\n" +
+                "\t\t\"userData1\": \"temp1\",\n" +
+                "\t\t\"userData2\": \"temp2\"\n" +
+                "\t},\n" +
+                "\t\"userDataInDatabase\": {\n" +
+                "\t\t\"userData\": \"value\"\n" +
+                "\t},\n" +
+                "\t\"customSigningKey\": \"string\",\n" +
+                "\t\"enableAntiCsrf\": false\n" +
+                "}";
+
+        JsonObject sessionBody = new JsonParser().parse(sessionJsonInput).getAsJsonObject();
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        // Create session
+        JsonObject session = io.supertokens.test.httpRequest.HttpRequest
+                .sendJsonPOSTRequest(process.getProcess(), "", "http://localhost:3567/recipe/session",
+                        sessionBody, 1000, 1000, null, Utils.getCdiVersionLatestForTests(), Session.RECIPE_ID);
+
+        assertEquals(session.get("status").getAsString(), "OK");
+
+        // Get session info
+        HashMap<String, String> map = new HashMap<>();
+        map.put("sessionHandle", session.get("session").getAsJsonObject().get("handle").getAsString());
+
+        JsonObject response = io.supertokens.test.httpRequest.HttpRequest
+                .sendGETRequest(process.getProcess(), "", "http://localhost:3567/recipe/session", map, 1000, 1000,
+                        null,
+                        Utils.getCdiVersionLatestForTests(), Session.RECIPE_ID);
+
+        // Validate response
+        assertEquals(response.get("status").getAsString(), "OK");
+        assertEquals(response.entrySet().size(), 6);
+
+        assertEquals(response.get("userDataInDatabase").getAsJsonObject(),
+                sessionBody.get("userDataInDatabase").getAsJsonObject());
+
+        assertEquals(response.get("userDataInJWT").getAsJsonObject(),
+                sessionBody.get("userDataInJWT").getAsJsonObject());
+
+        assertEquals(response.get("userId").getAsString(),
+                sessionBody.get("userId").getAsString());
+
+        assertNotNull(response.get("expiry"));
+        assertNotNull(response.get("timeCreated"));
```

Comment:
May want to check that the type of these are `numbers`

Reply:
Done